src/languages/snowflake/snowflake.formatter.ts

```diff
@@ -0,0 +1,282 @@
+  'DROP STAGE',
+  'DROP STREAM',
+  'DROP TABLE',
+  'DROP TAG',
+  'DROP TASK',
+  'DROP USER',
+  'DROP VIEW',
+  'DROP WAREHOUSE',
+  'EXECUTE IMMEDIATE',
+  'EXECUTE TASK',
+  'EXPLAIN',
+  'GET',
+  'GRANT OWNERSHIP',
+  'GRANT ROLE',
+  'INSERT',
+  'LIST',
+  'MERGE',
+  'PUT',
+  'REMOVE',
+  'REVOKE ROLE',
+  'ROLLBACK',
+  'SELECT',
+  'SET',
+  'SHOW COLUMNS',
+  'SHOW CONNECTIONS',
+  'SHOW DATABASES',
+  'SHOW DATABASES IN FAILOVER GROUP',
+  'SHOW DATABASES IN REPLICATION GROUP',
+  'SHOW DELEGATED AUTHORIZATIONS',
+  'SHOW EXTERNAL FUNCTIONS',
+  'SHOW EXTERNAL TABLES',
+  'SHOW FAILOVER GROUPS',
+  'SHOW FILE FORMATS',
+  'SHOW FUNCTIONS',
+  'SHOW GLOBAL ACCOUNTS',
+  'SHOW GRANTS',
+  'SHOW INTEGRATIONS',
+  'SHOW LOCKS',
+  'SHOW MANAGED ACCOUNTS',
+  'SHOW MASKING POLICIES',
+  'SHOW MATERIALIZED VIEWS',
+  'SHOW NETWORK POLICIES',
+  'SHOW OBJECTS',
+  'SHOW ORGANIZATION ACCOUNTS',
+  'SHOW PARAMETERS',
+  'SHOW PIPES',
+  'SHOW PRIMARY KEYS',
+  'SHOW PROCEDURES',
+  'SHOW REGIONS',
+  'SHOW REPLICATION ACCOUNTS',
+  'SHOW REPLICATION DATABASES',
+  'SHOW REPLICATION GROUPS',
+  'SHOW RESOURCE MONITORS',
+  'SHOW ROLES',
+  'SHOW ROW ACCESS POLICIES',
+  'SHOW SCHEMAS',
+  'SHOW SEQUENCES',
+  'SHOW SESSION POLICIES',
+  'SHOW SHARES',
+  'SHOW SHARES IN FAILOVER GROUP',
+  'SHOW SHARES IN REPLICATION GROUP',
+  'SHOW STAGES',
+  'SHOW STREAMS',
+  'SHOW TABLES',
+  'SHOW TAGS',
+  'SHOW TASKS',
+  'SHOW TRANSACTIONS',
+  'SHOW USER FUNCTIONS',
+  'SHOW USERS',
+  'SHOW VARIABLES',
+  'SHOW VIEWS',
+  'SHOW WAREHOUSES',
+  'TRUNCATE MATERIALIZED VIEW',
+  'TRUNCATE TABLE',
+  'UNDROP DATABASE',
+  'UNDROP SCHEMA',
+  'UNDROP TABLE',
+  'UNDROP TAG',
+  'UNSET',
+  'UPDATE',
+  'USE DATABASE',
+  'USE ROLE',
+  'USE SCHEMA',
+  'USE SECONDARY ROLES',
+  'USE WAREHOUSE',
+]);
+
+const reservedSetOperations = expandPhrases(['UNION [ALL]', 'MINUS', 'EXCEPT', 'INTERSECT']);
+
+const reservedJoins = expandPhrases([
+  '[INNER] JOIN',
+  '{LEFT | RIGHT | FULL} [OUTER] JOIN',
+  '{CROSS, NATURAL} JOIN',
+  'NATURAL {LEFT | RIGHT | FULL} [OUTER] JOIN',
+]);
+
+// TODO: unsure what belongs in here...
+const reservedPhrases = expandPhrases([]);
+
+export default class SnowflakeFormatter extends Formatter {
+  tokenizer() {
+    return new Tokenizer({
+      reservedCommands,
+      reservedSelect,
+      reservedSetOperations,
+      reservedJoins,
+      // https://docs.snowflake.com/en/sql-reference/functions/case.html
+      reservedDependentClauses: ['WHEN', 'ELSE'],
+      reservedPhrases,
+      reservedKeywords: keywords,
+      reservedFunctionNames: functions,
+      // TODO $$ is allowed, but also ' and ` how to achieve this?
+      // https://docs.snowflake.com/en/sql-reference/data-types-text.html#string-constants
+      // Backticks only allowed for inline Javascript:
+      // https://docs.snowflake.com/en/sql-reference/stored-procedures-javascript.html#line-continuation
+      stringTypes: ['$$', `'`, '`'],
```

Comment:
As I understand, backticks (`` ` ``) aren't actually part of Snowflake SQL syntax (the docs suggest their usage inside JavaScript when dealing with SQL strings).